src/components/RegisterPage/RegisterForm.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import submitConsents from 'api/Customer/submitConsents';
+import Loader from 'components/Loader';
+import Consent from 'components/Consents';
+import { FromStyled, FormErrorStyled } from 'components/LoginPage/LoginStyled';
+import Button from 'components/Button';
+import EmailInput from 'components/EmailInput';
+import PasswordInput from 'components/PasswordInput';
+import {
+  validateRegisterPassword,
+  validateEmailField,
+  validateConsentsField
+} from 'util/validators';
+import { selectPublisherConfig } from 'appRedux/publisherConfigSlice';
+import { selectPublisherConsents } from 'appRedux/publisherConsentsSlice';
+import registerCustomer from 'api/Auth/registerCustomer';
+import getCustomerLocales from 'api/Customer/getCustomerLocales';
+import Auth from 'services/auth';
+import { useAppSelector } from 'appRedux/store';
+import { Consent as ConsentType } from 'types/Consents.types';
+import { RegisterFormProps } from './RegisterForm.types';
+
+type Errors = {
+  email: string;
+  password: string;
+  consents: string;
+};
+
+const errorsInitialState = {
+  email: '',
+  password: '',
+  consents: ''
+};
+
+export const RegisterForm = ({ onSuccess }: RegisterFormProps) => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [consents, setConsents] = useState<boolean[]>([]);
+  const [errors, setErrors] = useState<Errors>(errorsInitialState);
+  const [generalError, setGeneralError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [consentDefinitions, setConsentDefinitions] = useState<ConsentType[]>(
+    []
+  );
+  const [processing, setProcessing] = useState(false);
+  const [disableActionButton, setDisableActionButton] = useState(false);
+
+  const { t } = useTranslation();
+  const { publisherId } = useAppSelector(selectPublisherConfig);
+  const { error: publisherConsentsError } = useAppSelector(
+    selectPublisherConsents
+  );
+
+  const handleClickShowPassword = () =>
+    setShowPassword((prevValue) => !prevValue);
+
+  const validateEmail = () => {
+    const message = validateEmailField(email);
+    setErrors((prevValue) => {
+      return {
+        ...prevValue,
+        email: message
+      };
+    });
+  };
+
+  const validatePassword = () => {
+    const message = validateRegisterPassword(password);
+    setErrors((prevValue) => {
+      return {
+        ...prevValue,
+        password: message
+      };
+    });
+  };
+
+  const validateFields = () => {
+    const errorFields = {
+      email: validateEmailField(email),
+      password: validateRegisterPassword(password),
+      consents: validateConsentsField(consents, consentDefinitions)
+    };
+    setErrors(errorFields);
+    return !(Object.keys(errorFields) as Array<keyof Errors>).find(
```

Comment:
I've simpliefied it a little and now it look like this:
`return !Object.values(errorFields).some((error) => error !== '');`
but it can be also simplified even further with  
`return !Object.values(errorFields).some(Boolean);`
but then for me its less readable thats why I leaved that first solution but if you want we can change it